### ISSUE TITLE
Uncaught Error: Cannot decorate a destroyed marker

### ISSUE THREAD

**Issue by kobayashicoding** [steelbrain/linter-ui-default#464]
**Atom**: 1.23.1 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.13.2
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.6.10

### Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At /Applications/Atom.app/Contents/Resources/app/src/decoration-manager.js:191

Error: Cannot decorate a destroyed marker
    at DecorationManager.decorateMarker (/Applications/Atom.app/Contents/Resources/app/src/decoration-manager.js:183:27)
    at TextEditor.decorateMarker (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2109:41)
    at Editor.decorateMarker (/packages/linter-ui-default/lib/editor/index.js:290:23)
    at _loop (/packages/linter-ui-default/lib/editor/index.js:283:12)
    at Editor.apply (/packages/linter-ui-default/lib/editor/index.js:261:61)
    at /packages/linter-ui-default/lib/editors.js:56:48
    at Array.forEach (native)
    at /packages/linter-ui-default/lib/editors.js:56:23
    at Array.forEach (native)
    at Editors.update (/packages/linter-ui-default/lib/editors.js:53:15)
    at LinterUI.render (/packages/linter-ui-default/lib/main.js:73:17)
    at /packages/linter/lib/ui-registry.js:32:16
    at Set.forEach (native)
    at UIRegistry.render (/packages/linter/lib/ui-registry.js:31:20)
    at /packages/linter/lib/main.js:187:23
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at MessageRegistry.update (/packages/linter/lib/message-registry.js:130:20)
    at later (/packages/linter/node_modules/sb-debounce/index.js:9:14)
```

### Commands

```
     -0:46.4.0 pane:show-item-3 (input.hidden-input)
     -0:39 core:paste (input.hidden-input)
  5x -0:38.3.0 core:undo (input.hidden-input)
     -0:30.6.0 editor:newline (input.hidden-input)
     -0:27.4.0 core:paste (input.hidden-input)
     -0:24.3.0 core:backspace (input.hidden-input)
     -0:20.7.0 core:copy (input.hidden-input)
     -0:19.9.0 core:paste (input.hidden-input)
  3x -0:16.2.0 core:save (input.hidden-input)
     -0:13.5.0 editor:consolidate-selections (input.hidden-input)
     -0:13.5.0 core:cancel (input.hidden-input)
     -0:13.4.0 editor:consolidate-selections (input.hidden-input)
     -0:13.4.0 core:cancel (input.hidden-input)
     -0:13.2.0 editor:consolidate-selections (input.hidden-input)
     -0:13.2.0 core:cancel (input.hidden-input)
  4x -0:04.8.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-clock 0.1.15 
autocomplete-modules 1.8.0 
autocomplete-paths 2.12.1 
base16-ocean-light-syntax 0.3.0 
busy-signal 1.4.3 
intentions 1.1.5 
language-babel 2.81.1 
language-vue 0.23.1 
linter 2.2.0 
linter-eslint 8.4.0 
linter-ui-default 1.6.10 
open-in-browser 0.5.2 
seti-icons 1.5.4 
teletype 0.4.0 
```

**Comment by dzeignyc** [comment]
A duplicate issue can be found [here](https://github.com/steelbrain/linter-ui-default/issues/434)

**Comment by kagzgenius** [comment]
**I have the same issue.**

**Atom**: 1.23.3 x64
**Electron**: 1.6.15
**OS**: Microsoft Windows 10 Enterprise
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.6.10


### Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At C:\Users\Matthew Obwaya\AppData\Local\atom\app-1.23.3\resources\app\src\decoration-manager.js:191

Error: Cannot decorate a destroyed marker
    at DecorationManager.decorateMarker (~/AppData/Local/atom/app-1.23.3/resources/app/src/decoration-manager.js:183:27)
    at TextEditor.decorateMarker (~/AppData/Local/atom/app-1.23.3/resources/app/src/text-editor.js:2109:41)
    at Editor.decorateMarker (/packages/linter-ui-default/lib/editor/index.js:290:23)
    at _loop (/packages/linter-ui-default/lib/editor/index.js:283:12)
    at Editor.apply (/packages/linter-ui-default/lib/editor/index.js:261:61)
    at /packages/linter-ui-default/lib/editors.js:56:48
    at Array.forEach (native)
    at /packages/linter-ui-default/lib/editors.js:56:23
    at Array.forEach (native)
    at Editors.update (/packages/linter-ui-default/lib/editors.js:53:15)
    at LinterUI.render (/packages/linter-ui-default/lib/main.js:73:17)
    at /packages/linter/lib/ui-registry.js:32:16
    at Set.forEach (native)
    at UIRegistry.render (/packages/linter/lib/ui-registry.js:31:20)
    at /packages/linter/lib/main.js:187:23
    at Function.module.exports.Emitter.simpleDispatch (~/AppData/Local/atom/app-1.23.3/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (~/AppData/Local/atom/app-1.23.3/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at MessageRegistry.update (/packages/linter/lib/message-registry.js:130:20)
    at later (/packages/linter/node_modules/sb-debounce/index.js:9:14)
```

### Commands

```
  9x -6:39.4.0 intentions:highlight (input.hidden-input)
     -6:38.6.0 core:select-all (input.hidden-input)
  8x -6:37.4.0 intentions:highlight (input.hidden-input)
     -6:36.7.0 core:copy (input.hidden-input)
 13x -6:34.5.0 intentions:highlight (input.hidden-input)
     -6:33.7.0 core:select-all (input.hidden-input)
     -6:33 core:paste (input.hidden-input)
     -6:32.4.0 intentions:highlight (input.hidden-input)
     -6:31.9.0 core:save (input.hidden-input)
 60x -6:30.3.0 intentions:highlight (input.hidden-input)
     -6:26.2.0 core:select-all (input.hidden-input)
     -6:25.2.0 core:paste (input.hidden-input)
  7x -6:23.8.0 intentions:highlight (input.hidden-input)
     -6:23.2.0 core:save (input.hidden-input)
     -6:21 intentions:highlight (input.hidden-input)
  4x -3:31.2.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.30.9 
autoclose-html 0.23.0 
busy-signal 1.4.3 
color-picker 2.2.5 
intentions 1.1.5 
language-babel 2.81.1 
linter 2.2.0 
linter-csslint 2.0.0 
linter-eslint 8.4.0 
linter-htmlhint 1.4.0 
linter-json-lint 0.1.2 
linter-ui-default 1.6.10 
teletype 0.6.0 
```

**Comment by yagudaev** [comment]
Any updates on this or workaround to prevent it from popping up? It is quite distracting when coding

**Comment by sompylasar** [comment]
For the record, the error is thrown from here:
https://github.com/steelbrain/linter-ui-default/blob/dca54d5b3b70eceeecf8499a3fd7a11aef6da703/lib/editor/index.js#L288-L294

The `marker` originates from here:
https://github.com/steelbrain/linter-ui-default/blob/dca54d5b3b70eceeecf8499a3fd7a11aef6da703/lib/editor/index.js#L268-L270

following by an `marker.onDidChange` call:
https://github.com/steelbrain/linter-ui-default/blob/dca54d5b3b70eceeecf8499a3fd7a11aef6da703/lib/editor/index.js#L273-L283

which may cause synchronous marker state change in the stack of observers:
https://github.com/atom/text-buffer/blob/222fc436659dfb3f2c6f52f7c3368f36c6c643ce/src/marker.coffee#L104-L109
```
  onDidChange: (callback) ->
    unless @hasChangeObservers
      @previousEventState = @getSnapshot(@getRange())
      @hasChangeObservers = true
      @layer.markersWithChangeListeners.add(this)
    @emitter.on 'did-change', callback
```

**Comment by prescottprue** [comment]
I was able to get this to go away by quitting and restarting atom.

**Comment by sompylasar** [comment]
@prescottprue This is a temporary relief, it comes back.

@yagudaev @kobayashicoding @kagzgenius 

Could y'all please try in your local `linter-ui-default/lib/editor/index.js` (`Atom Settings -> Open Config Folder -> packages -> linter-ui-default`) to reorder the lines so that `this.decorateMarker(message, marker)` happens before `marker.onDidChange(` and see if relief is more permanent? I can't tell for sure yet.

```diff
+      this.decorateMarker(message, marker)
       marker.onDidChange(({ oldHeadPosition, newHeadPosition, isValid }) => {
         if (!isValid || (newHeadPosition.row === 0 && oldHeadPosition.row !== 0)) {
           return
         }
         if (message.version === 1) {
           message.range = marker.previousEventState.range
         } else {
           message.location.position = marker.previousEventState.range
         }
       })
-      this.decorateMarker(message, marker) 
```

**Comment by callmeaponte** [comment]
@sompylasar I tried reordering those lines. The error went away for a couple of days... but now it's back again :(

**Comment by Dajust** [comment]
I can't even write jack anymore, the thing pops up on every keystroke. How do we roll back to an old working version of this package, any ideas?

**Comment by damianobarbati** [comment]
+1! what's going on?

**Comment by Dajust** [comment]
I think @sompylasar fix is working for me so far! Thanks man :pray:

**Comment by gplanansky** [comment]
likewise @sompylasar fix working so far

**Comment by TheFullResolution** [comment]
@sompylasar fix works for me too 👍

**Comment by amcvitty** [comment]
Seems to be working for me too

**Comment by steelbrain** [comment]
Since this change here has been proven by many, I'm gonna include it in the next update

**Comment by steelbrain** [comment]
Presuming fixed as of v1.7.0
